src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,47 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  static addAnimal(animal) {
+    this.alive.push(animal);
+  }
+
+  static removeAnimal(animal) {
+    this.alive.splice(this.alive.indexOf(animal.name), 1);
+  }
```

Comment:
Let's reduce complexity and apply filter here.
splice(indexOf(...)) -> n ** 2
filter(...) -> n

Reply:
Yes, I agree. It's gonna be a nice piece of the optimization of performance. Thank you :)